packages/user/src/models/invite.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  BeforeCreate,
+  Entity,
+  OnInit,
+  PrimaryKey,
+  Property,
+  Index,
+  Filter,
+  OptionalProps,
+} from "@mikro-orm/core"
+
+import { DALUtils, generateEntityId } from "@medusajs/utils"
+import { DAL } from "@medusajs/types"
+
+type OptionalFields =
+  | "metadata"
+  | "accepted"
+  | DAL.SoftDeletableEntityDateColumns
+@Entity({ tableName: "invite" })
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
+export default class Invite {
+  [OptionalProps]: OptionalFields
+
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @Index({
+    name: "invite_user_identifier_index",
+    expression: `create unique index "IDX_invite_email" on "invite" ("email") where deleted_at is null;`,
```

Comment:
todo: I suggest you to use this [utils](https://github.com/medusajs/medusa/blob/develop/packages/utils/src/common/create-psql-index-helper.ts) as mentioned in slack. examples [here](https://github.com/medusajs/medusa/blob/develop/packages/fulfillment/src/models/fulfillment-item.ts#L23)